### DISCUSSION TITLE
Support for select count(*) in `builder/for-query`

### DISCUSSION THREAD

**Discussion by slipset** [seancorfield/next-jdbc#155] [Ideas]
Problem: I'd like to do a `select count(*) from foo where bar = baz`, but the current implementation of `builder/for-query` doesn't have way to do that, as far as I can see.

Would you be interested in a patch where you could pass `{:count true}`  in the opts map to make `builder/for-query` generate `SELECT count(*) ...` rather than just `SELECT *`?

**Comment by slipset** [comment]
For posterity, I RTFS'ed a bit and saw that by passing
`{:columns [["count(*)" :total]]}` I can do what I want.

**Reply by seancorfield** [reply]
This is unsupported and out of scope : my official answer is "use HoneySQL for this". The friendly SQL functions in `next.jdbc` are deliberately very simple and will not be expanded.

That said, I'm glad you found a way to make it work 😁
